sys-utils/lsns.c

```diff
@@ -721,6 +836,32 @@ static int read_namespaces(struct lsns *ls)
 					}
 				}
 			}
+
+			/* lsns scans /proc/[0-9]+ for finding namespaces.
+			 * So if a namespace has no process, lsns cannot
+			 * find it. Here we call it a missing namespace.
+			 *
+			 * If the id for a related namesspce is known but
+			 * namespace for the id is not found, there must
+			 * be orphan namespaces. A missing namespace is an
+			 * owner or a parent of the orphan namespace.
+			 */
+			for (rela = 0; rela < MAX_RELA; rela++) {
+				if (ns->related_id[rela] != 0
+				    && ns->related_ns[rela] == NULL) {
+					ns->related_ns[rela] = orphan[rela];
+					orphan[rela] = ns;
+				}
+			}
+		}
+	}
+
+	for (rela = 0; rela < MAX_RELA; rela++) {
+		while (orphan[rela]) {
+			struct lsns_namespace *current = orphan[rela];
+			orphan[rela] = orphan[rela]->related_ns[rela];
+			current->related_ns[rela] = NULL;
+			interpolate_missing_namespaces (ls, current, rela);
```

Comment:
So, if I good understand the idea then you use orphan[]->related_ns as a list of orphans, right?

Reply:
Yes. I abused the related_ns fields to record orphan namespaces to avoid adding fields to lsns_namespace structure only for the purpose.